src/forum/forum_post.rs

```diff
@@ -162,9 +136,58 @@ impl ForumPost {
             parent_id: Some(parent_id),
             author,
             created_at,
+            deleted_at: None,
             message,
         };
 
         Ok(forum_reply)
     }
+
+    pub fn soft_delete_post(conn: &rusqlite::Connection, id: usize) -> ForumResult<()> {
+        let deleted_at = chrono::Local::now().naive_local();
+        conn.execute(
+            "UPDATE forum_posts SET deleted_at = ?1 WHERE id = ?2",
+            (deleted_at, id),
+        )
+        .map_err(ForumError::DatabaseError)?;
+        Ok(())
```

Comment:
What if id doesn't exist?

Can we see how many affected rows the query has and fail on 0?

Reply:
I've added an IF to ensure deleted_at is null, and then a check with an error if it has already been deleted.

Reply:
Excellent. Now we get a `NotFound` when deleting something deleted. Sounds fine.

More importantly, the timestamp doesn't push forward every time a post gets deleted again.